invenio_queues/__init__.py

```diff
@@ -6,7 +6,149 @@
 # Invenio is free software; you can redistribute it and/or modify it
 # under the terms of the MIT License; see LICENSE file for more details.
 
-"""Invenio module for collecting statistics."""
+
+r"""Invenio module for managing queues.
+
+This guide will show you how to get started with Invenio-Queues. It assumes
+that you already have knowledge of Flask applications and Invenio modules.
+
+It will then explain key topics and concepts of this module.
+
+Getting started
+---------------
+
+You will learn how to register queues and to interact with it. To begin with,
+we assume that you already have setup your virtual environment and install
+the module:
+
+You now need an application to work with, that can be created with the
+following commands in a Python shell:
+
+.. code-block:: python
+
+   from flask import Flask
+   app = Flask('myapp')
+
+You can add the queue in the configuration:
+
+.. code-block:: python
+
+   app.config['QUEUES_DEFINITIONS'] = [
+       {
+           'name': 'notifications',
+           'exchange': Exchange(
+               'example',
+               type='direct',
+               delivery_mode='transient',  # in-memory queue
+           ),
+       },
+   ]
+
+You can then initialize the module:
+
+.. code-block:: python
+
+   from invenio_queues.ext import InvenioQueues
+   ext_queues = InvenioQueues(app)
+
+In our example, we are using RabbitMQ as a broker, which can be configured
+as follow::
+
+    app.config['QUEUES_BROKER_URL'] = 'amqp://localhost:5672'
+
+Register queues
+^^^^^^^^^^^^^^^
+
+To register queues, you need to start by creating an exchange
+(https://docs.celeryproject.org/projects/kombu/en/stable/reference/kombu.html#exchange)
+for the queues:
+
+.. code-block:: python
+
+   from kombu import Exchange
+
+   default_exchange = Exchange(
+       'example',
+       type='direct',
+       delivery_mode='transient',  # in-memory queue
+   )
+
+You can now configure the queues as followed:
+
+.. code-block:: python
+
+   from invenio_queues.proxies import current_queues
+   from invenio_queues.queue import Queue
+
+   connection_pool = app.config.get('QUEUES_CONNECTION_POOL')
+
+   ext_queues.queues['notifications'] = Queue(
+       default_exchange,
+       'notifications',
+       connection_pool
+   )
+
+   ext_queues.queues['jobs'] = Queue(
+       default_exchange,
+       'jobs',
+       connection_pool
+   )
```

Comment:
This part can be removed, since we now use the `QUEUES_DEFINITIONS` config

Reply:
Do not you want to allow the two different mechanisms?

Reply:
This way is not meant to be used by actual users of the library, it was just a way to quickly create/test queues locally while writing the docs.

There should only be two mechanisms available:

- `QUEUES_DEFINITIONS` config
- `invenio_queues.queues` entrypoint